index.js

```diff
@@ -37,12 +37,33 @@ class Schematic {
     return this.Block.fromStateId(this.getBlockStateId(pos), 0)
   }
 
+  static async copy (world, start, end, offset, version) {
+    const size = end.minus(start).offset(1, 1, 1)
+    const palette = []
+    const blocks = []
+    const cursor = new Vec3(0, 0, 0)
+    for (cursor.y = start.y; cursor.y <= end.y; cursor.y++) {
+      for (cursor.z = start.z; cursor.z <= end.z; cursor.z++) {
+        for (cursor.x = start.x; cursor.x <= end.x; cursor.x++) {
+          const stateId = await world.getBlockStateId(cursor)
```

Comment:
This could be made faster by doing all columns concurrently from the other ones
(This implementation will read region files one by one whereas they could all be read at once)
To do that it would probably make sense to have an iterator in pworld

Reply:
Or we can just give the world.sync instance. Its what we have in mineflayer and what I use in my examples. I added the await to be compatible with async world (even if slower)

Reply:
In my tests, saving a 20x20x20 schematic is pretty much instant.

Reply:
Would be interesting to measure it. Is it 1m or 100ms ?
One use case of this function might be to extract a bunch of part of worlds as schematics from a world, so it's useful if it's a bit faster I think

Reply:
A 37x26x22 schematic took 27ms to copy. This could be later improved, but for current use cases, it should be fast enough.